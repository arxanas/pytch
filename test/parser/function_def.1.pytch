let print_fst(x, y) =
  print(x)

let print_snd(x, y) = print(y)
