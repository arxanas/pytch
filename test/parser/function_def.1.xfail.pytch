def print_fst(x, y) =>
  print(x)

def print_snd(x, y) => print(y)
