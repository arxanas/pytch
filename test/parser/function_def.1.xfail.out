SyntaxTree
    DefExpr
        Token 'def'
        Token 'print_fst'
        ParameterList
            Token '('
            Parameter
                VariablePattern
                    Token 'x'
                Token ','
            Parameter
                VariablePattern
                    Leading ' '
                    Token 'y'
                <missing>
            Token ')'
        Leading ' '
        Token '=>'
        Trailing '\n'
        FunctionCallExpr
            IdentifierExpr
                Leading '  '
                Token 'print'
            ArgumentList
                Token '('
                Argument
                    IdentifierExpr
                        Token 'x'
                    <missing>
                Token ')'
                Trailing '\n'
                Trailing '\n'
        Token DUMMY_IN_FOR_LET ''
        DefExpr
            Token 'def'
            Token 'print_snd'
            ParameterList
                Token '('
                Parameter
                    VariablePattern
                        Token 'x'
                    Token ','
                Parameter
                    VariablePattern
                        Leading ' '
                        Token 'y'
                    <missing>
                Token ')'
            Leading ' '
            Token '=>'
            FunctionCallExpr
                IdentifierExpr
                    Leading ' '
                    Token 'print'
                ArgumentList
                    Token '('
                    Argument
                        IdentifierExpr
                            Token 'y'
                        <missing>
                    Token ')'
                    Trailing '\n'
            Token DUMMY_IN_FOR_LET ''
            <missing>
    Token EOF ''
